pkg/promtail/targets/gcplog/formatter.go

```diff
@@ -53,6 +58,7 @@ func format(
 	// mandatory label for gcplog
 	lbs := labels.NewBuilder(nil)
 	lbs.Set("resource_type", ge.Resource.Type)
+	lbs.Set("promtail_instance", instanceID.String())
```

Comment:
I think some explanation here of _why_ we've put this unique instance ID in place will be helpful.
I also think `promtail_instance` might be somewhat misleading but I can't think of a clearer name right now - perhaps something that captures its purpose a bit more, and/or documentation to describe its purpose.

Reply:
Added comment and doc explaining the rationale!

Regarding name, first I thought may be `instance_id` would be good fit. but its too good that original log entries may actually have it in future!. So just went with `promtail_instance`. If you have any better name feel free :)

Reply:
Thanks for adding this! I don't think we should duplicate the docs in case they go out of sync - simply linking them with a very brief explanation should suffice.

Perhaps something like this:

```golang
// We add a `promtail_instance` label to avoid identical streams with different timestamps if the `use_incoming_timestamp` option is set to false, which can lead to out-of-order errors
```